images/linux/scripts/installers/pypy.sh

```diff
@@ -74,14 +74,12 @@ function InstallPyPy
 }
 
 # Installation PyPy
-uri="https://downloads.python.org/pypy/"
-download_with_retries $uri "/tmp" "pypyUrls.html" compressed
-pypyVersions="$(cat /tmp/pypyUrls.html | grep 'linux64' | awk -v uri="$uri" -F'>|<' '{print uri$5}')"
+pypyVersions=$(curl https://downloads.python.org/pypy/versions.json)
 toolsetVersions=$(get_toolset_value '.toolcache[] | select(.name | contains("PyPy")) | .versions[]')
 
 for toolsetVersion in $toolsetVersions; do
-    latestMajorPyPyVersion=$(echo "${pypyVersions}" | grep -E "pypy${toolsetVersion}-v[0-9]+\.[0-9]+\.[0-9]+-" | head -1)
-
+    latestMajorPyPyVersion=$(echo $pypyVersions | jq --arg toolsetVersion $toolsetVersion '.[] | select((.python_version | contains($toolsetVersion)) and .stable == true).files[] | select(.arch == "x64" and .platform == "linux").download_url' | head -1)
+    latestMajorPyPyVersion=`sed -e 's/^"//' -e 's/"$//' <<<"$latestMajorPyPyVersion"`
```

Comment:
remove this command and add  a param -r to jq

Reply:
Maybe I'm doing smth wrong but "-r" not working locally
`jq: error: Could not open file .[] | select((.python_version | contains($toolsetVersion)) and .stable == true).files[] | select(.arch == "x64" and .platform == "linux").download_url: No such file or directory`

Reply:
```
:~$ echo $pypyVersions | jq -r --arg toolsetVersion $toolsetVersion '.[] | select((.python_version | contains($toolsetVersion)) and .stable == true).files[] | select(.arch == "x64" and .platform == "linux").download_url' | head -1
https://downloads.python.org/pypy/pypy3.8-v7.3.7-linux64.tar.bz2
```

Reply:
Updated